docs/learn/prompts.md

```diff
@@ -345,7 +388,7 @@ It leverages Pydantic's [`BaseModel`](https://docs.pydantic.dev/latest/concepts/
 
 To recap, you can create a prompt using `BasePrompt` by defining a class that inherits from it:
```

Comment:
i'm realizing that this entire section on `BasePrompt` is out of place. For example, `computed_field` should really be discussed in dynamic configuration alongside the section where we discuss computed fields for call functions.

the `run` method should be discussed in the "Calls" documentation alongside where we should making a call.

Essentially this documentation should really just focus on the core features of prompt templates and prompts and push everything else into the respective section where it makes sense